src/sentry/static/sentry/app/views/releases/detail/withCommitRepos.tsx

```diff
@@ -0,0 +1,217 @@
+import React from 'react';
+import {RouteComponentProps} from 'react-router';
+import * as Sentry from '@sentry/react';
+
+import {addErrorMessage} from 'app/actionCreators/indicator';
+import {Client} from 'app/api';
+import Button from 'app/components/button';
+import {Body, Main} from 'app/components/layouts/thirds';
+import LoadingIndicator from 'app/components/loadingIndicator';
+import {IconCommit, IconWarning} from 'app/icons';
+import {t} from 'app/locale';
+import {Organization, Repository} from 'app/types';
+import getDisplayName from 'app/utils/getDisplayName';
+import withApi from 'app/utils/withApi';
+import withOrganization from 'app/utils/withOrganization';
+import withRepositories from 'app/utils/withRepositories';
+import EmptyMessage from 'app/views/settings/components/emptyMessage';
+
+import {ReleaseContext} from '.';
+
+// These props are required when using this HoC
+type DependentProps = RouteComponentProps<{orgId: string; release: string}, {}>;
+
+type HoCsProps = {
+  api: Client;
+  organization: Organization;
+  repositories?: Repository[];
+  repositoriesLoading?: boolean;
+  repositoriesError?: Error;
+};
+
+type State = {
+  commitRepositories: Repository[];
+  isLoading: boolean;
+  activeCommitRepository?: Repository;
+};
+
+const withCommitRepos = <P extends DependentProps>(
+  WrappedComponent: React.ComponentType<P>
+) =>
+  withApi(
+    withOrganization(
+      withRepositories(
```

Comment:
Will this result in chained network calls? Should the component using this HoC apply `withRepositories()` instead? That would let it fire both requests in parallel and render the correct state once both responses have been returned.

Reply:
only `withRepositories` and` withReleaseRepos` will fire requests and `withReleaseRepos` will only send a request if  response of the first request contains an array